python/py-dateutil/Portfile

```diff
@@ -39,7 +43,8 @@ if {${name} ne ${subport}} {
 
     depends_test-append \
                     port:py${python.version}-freezegun \
-                    port:py${python.version}-hypothesis
+                    port:py${python.version}-hypothesis \
+                    port:py${python.version}-pytest-cov
```

Comment:
this is not correct since we don't measure coverage with the `pytest-cov` plugin. The `python` PortGroup adds `"-o addopts=''"` as test arguments and thus coverage will not be run. Please remove.

Reply:
@reneeotten https://github.com/macports/macports-ports/pull/26391#discussion_r1830319620:
> this is not correct since we don't measure coverage with the `pytest-cov` plugin. The `python` PortGroup adds `"-o addopts=''"` as test arguments and thus coverage will not be run. Please remove.

What you suggest doesn’t work without a patch, because of [this](https://github.com/dateutil/dateutil/blob/2.9.0.post0/tests/conftest.py#L20).

```
:info:test Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-dateutil/py313-dateutil/work/python-dateutil-2.9.0.post0" && py.test-3.13 -o addopts='' 
:debug:test system:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-dateutil/py313-dateutil/work/python-dateutil-2.9.0.post0" && py.test-3.13 -o addopts='' 
:info:test ============================= test session starts ==============================
:info:test platform darwin -- Python 3.13.0, pytest-7.4.4, pluggy-1.5.0
:info:test rootdir: /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-dateutil/py313-dateutil/work/python-dateutil-2.9.0.post0
:info:test configfile: setup.cfg
:info:test plugins: hypothesis-6.115.3
:info:test collected 2096 items
:info:test INTERNALERROR> Traceback (most recent call last):
[…]
:info:test INTERNALERROR>   File "/opt/local/var/macports/build/_opt_local_var_m
acports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-dat
eutil/py313-dateutil/work/python-dateutil-2.9.0.post0/tests/conftest.py", line 20, in pytest_collection_modifyitems
:info:test INTERNALERROR>     item.add_marker(pytest.mark.no_cover)
:info:test INTERNALERROR>                     ^^^^^^^^^^^^^^^^^^^^
:info:test INTERNALERROR>   File "/opt/local/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/_pytest/mark/structures.py", line 549, in __getattr__
:info:test INTERNALERROR>     warnings.warn(
:info:test INTERNALERROR>     ~~~~~~~~~~~~~^
:info:test INTERNALERROR>         "Unknown pytest.mark.%s - is this a typo?  You can register "
:info:test INTERNALERROR>         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
:info:test INTERNALERROR>     ...<3 lines>...
:info:test INTERNALERROR>         2,
:info:test INTERNALERROR>         ^^
:info:test INTERNALERROR>     )
:info:test INTERNALERROR>     ^
:info:test INTERNALERROR> pytest.PytestUnknownMarkWarning: Unknown pytest.mark.no_cover - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
:info:test ============================ no tests ran in 0.71s =============================
:info:test Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-dateutil/py313-dateutil/work/python-dateutil-2.9.0.post0" && py.test-3.13 -o addopts='' 
:info:test Exit code: 3
:error:test Failed to test py313-dateutil: command execution failed
```

Given the choice between patching and adding a test-only dependency, I chose the test-only dependency.

Reply:
but that test dependency you added is not available for many of the Python subports and we actually don't use the `pytest-cov` package when running the test-suite; so please do patch instead.

Reply:
@reneeotten https://github.com/macports/macports-ports/pull/26391#discussion_r1831720167:
> but that test dependency you added is not available for many of the Python subports and we actually don't use the `pytest-cov` package when running the test-suite; so please do patch instead.

OK, switched to the patch.